lib/pubsub/topic.js

```diff
@@ -63,6 +63,27 @@ function Topic(pubsub, options) {
 }
 
 /**
+ * Format a message object as the upstream API expects it.
+ *
+ * @private
+ *
+ * @return {object}
+ */
+Topic.formatMessage_ = function(message) {
+  if (!message.data) {
+    message = {
+      data: new Buffer(message).toString('base64')
+    };
+  }
+
+  if (!util.is(message.data, 'string') && !util.is(message.data, 'buffer')) {
+    message.data = JSON.stringify(message.data);
```

Comment:
Would be nice to have a `topic#formatMessage_` and whatever its inverse is. It seems the inverse logic is in `subscription#formatMessage_`, I believe. If those methods could be separated and tested together, that might simplify the pain we have here with understanding the different cases. Overall I think it should work because `JSON.parse(JSON.stringify(2))` is `2` but would be nice to let the tests tell us.